jaeger-core/src/main/java/com/uber/jaeger/baggage/RemoteBaggageRestrictionManager.java

```diff
@@ -0,0 +1,136 @@
+
+import static com.uber.jaeger.baggage.Constants.DEFAULT_MAX_VALUE_LENGTH;
+
+import com.uber.jaeger.baggage.http.BaggageRestriction;
+import com.uber.jaeger.exceptions.BaggageRestrictionErrorException;
+import com.uber.jaeger.metrics.Metrics;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class RemoteBaggageRestrictionManager implements BaggageRestrictionManager {
+  private static final int DEFAULT_REFRESH_INTERVAL_MS = 60000;
+  private static final int DEFAULT_REFRESH_DELAY_MS = 0;
+
+  private final String serviceName;
+  private final BaggageRestrictionProxy proxy;
+  private final Timer pollTimer;
+  private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+  private final Metrics metrics;
+  private final boolean denyBaggageOnInitializationFailure;
+  private boolean initialized;
+  private Map<String, Integer> restrictions = new HashMap<String, Integer>();
+
+  public RemoteBaggageRestrictionManager(
+      String serviceName, BaggageRestrictionProxy proxy, Metrics metrics, boolean denyBaggageOnInitializationFailure) {
+    this(serviceName, proxy, metrics, denyBaggageOnInitializationFailure,
+        DEFAULT_REFRESH_INTERVAL_MS, DEFAULT_REFRESH_DELAY_MS);
+  }
+
+  public RemoteBaggageRestrictionManager(
+      String serviceName,
+      BaggageRestrictionProxy proxy,
+      Metrics metrics,
+      boolean denyBaggageOnInitializationFailure,
+      int refreshIntervalMs,
+      int refreshDelayMs
+  ) {
+    this.serviceName = serviceName;
+    this.proxy = proxy;
+    this.metrics = metrics;
+    this.denyBaggageOnInitializationFailure = denyBaggageOnInitializationFailure;
+    this.initialized = false;
+
+    pollTimer = new Timer(true); // true makes this a daemon thread
+    pollTimer.schedule(
+        new TimerTask() {
+          @Override
+          public void run() {
+            updateBaggageRestrictions();
+          }
+        },
+        refreshDelayMs,
+        refreshIntervalMs);
+  }
+
+  void updateBaggageRestrictions() {
+    List<BaggageRestriction> response;
+    try {
+      response = proxy.getBaggageRestrictions(serviceName);
+    } catch (BaggageRestrictionErrorException e) {
+      metrics.baggageRestrictionsUpdateFailure.inc(1);
+      return;
+    }
+
+    updateBaggageRestrictions(response);
+    metrics.baggageRestrictionsUpdateSuccess.inc(1);
+  }
+
+  private void updateBaggageRestrictions(List<com.uber.jaeger.baggage.http.BaggageRestriction> restrictions) {
+    Map<String, Integer> baggageRestrictions = new HashMap<String, Integer>();
+    for (com.uber.jaeger.baggage.http.BaggageRestriction restriction : restrictions) {
+      baggageRestrictions.put(restriction.getBaggageKey(), restriction.getMaxValueLength());
+    }
+    ReentrantReadWriteLock.WriteLock writeLock = this.lock.writeLock();
+    writeLock.lock();
```

Comment:
I'm not convinced that we need synchronization because we are using `baggageRestrictions` like an Immutable Object. 
Simply declaring `this.restrictions` and `this.initialized` to be volatile should be sufficient. 
See https://docs.oracle.com/javase/tutorial/essential/concurrency/atomic.html